src/util/Repeat.cpp

```diff
@@ -17,17 +17,22 @@
 */
 //==============================================================================
 
-#pragma once
+#include "util/Repeat.hpp"
 
-#include <cstddef>
-#include <functional>
+#include <boost/asio/io_context.hpp>
 
-struct FakeAmendmentBlockAction {
-    std::reference_wrapper<std::size_t> callCount;
+namespace util {
 
-    void
-    operator()() const
-    {
-        ++(callCount.get());
-    }
-};
+Repeat::Repeat(boost::asio::io_context& ioc) : ioc_{ioc}, timer_(ioc_)
```

Comment:
nit: it's generally recommended to avoid using member variables to initialize other member variables in the init list. in this case it's fine but perhaps it's better to `timer_(ioc)` since it's possible